lib/models/app_state.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:convert';
+import 'package:meta/meta.dart';
+
+import './models.dart';
+
+@immutable
+class AppState {
+  final User user;
+
+  const AppState({
+    @required this.user,
+  });
+
+  factory AppState.initial() {
+    return AppState(
+      user: User.initial(),
+    );
+  }
+
+  AppState copyWith({@required User user}) {
+    return AppState(user: user ?? this.user);
+  }
+
+  static AppState fromJson(dynamic json) => 
+    AppState(user: User.fromJson(json["user"]));
+  
+  dynamic toJson() => {
```

Comment:
Do we have to convert the user and other objects in the global state to a JSON format?

Reply:
Yep we'll have to do convert the classes to JSON to save them in the redux store! and then convert them back to their class format (`fromJson`) when accessing them!